python-avd/pyavd/_eos_designs/structured_config/base/utils.py

```diff
@@ -76,3 +83,66 @@ def _build_source_interfaces(
             )
 
         return source_interfaces
+
+    def _get_vrf_and_source_interface(
+        self: AvdStructuredConfigBaseProtocol, vrf_input: str | None, vrfs: T_ProtocolVrfs, set_source_interfaces: bool, context: str
+    ) -> tuple[str, str | None]:
+        """
+        Helper function to interpret the VRF field for a management protocol.
+
+        If not set, the VRF is automatically picked up from the global setting `default_mgmt_method`.
+        The value of `vrf` will be interpreted according to these rules:
+        - `use_mgmt_interface_vrf` will return `(<mgmt_interface_vrf>, <vrfs[].source_interface or mgmt_interface>)`.
+          An error will be raised if `mgmt_ip` or `ipv6_mgmt_ip` are not configured for the device.
+        - `use_inband_mgmt_vrf` will return `(<inband_mgmt_vrf>, <vrfs[].source_interface or inband_mgmt_interface>)`.
+          An error will be raised if inband management is not configured for the device.
+        - Any other string will return `(<vrf_input>, None)`
```

Comment:
any reason to always return None if there is an interface in vrfs?

Reply:
actually this won't return None in that case so just need to update teh docstring

Reply:
```suggestion
        - Any other string will return `(<vrf_input>, <vrfs[].source_interface or None)`
```